osh/hub/osh_xmlrpc/scan.py

```diff
@@ -237,30 +237,36 @@ def find_tasks(request, query):
      * 'nvr': search by specific NVR
      * 'regex': find by provided regex (this is not match, but find, if you
                 want match, change your regex to "^<regex>$")
+     * 'comment': string, search by comment
+
+    Query also supports following optional keys:
+     * 'states': list, search by task states
 
     Returned list is ordered by date, when task finished -- latest task is
     first. Unfinished tasks are at the tail. If there is any problem with
     querying, empty list is returned.
     """
     if not isinstance(query, dict):
         return []
-    nvr = query.get('nvr')
     package_name = query.get('package_name')
-    comment = query.get('comment')
+    nvr = query.get('nvr')
     regex = query.get('regex')
+    comment = query.get('comment')
     states = query.get('states')
     latest = query.get('latest')
 
     result = []
-    tasks = None
     if nvr:
         tasks = Task.objects.filter(label=nvr)
-    if package_name:
+    elif package_name:
         tasks = Task.objects.filter(label__regex=package_name + r"-\d")
-    elif comment:
-        tasks = Task.objects.filter(comment__regex=comment)
     elif regex:
         tasks = Task.objects.filter(label__regex=regex)
+    elif comment:
+        tasks = Task.objects.filter(comment__regex=comment)
+    else:
+        tasks = Task.objects.none()
+
     if states:
         tasks = tasks.filter(state__in=states)
     if tasks is not None:
```

Comment:
This check is now redundant because `tasks` will never be set to `None`.

Reply:
updated